dpytools/validation/json/validation.py

```diff
@@ -0,0 +1,68 @@
+import json
+from pathlib import Path
+from typing import Dict, Optional, Union
+from urllib.parse import urlparse
+
+import jsonschema
+from jsonschema import ValidationError
+
+"""
+# data_dict is for passing in a dictionary to be validated
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary)
+
+# msg is to include a helpful context when debugging (i.e "what we're validating")
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary, msg="Some helpful message should this validation fail")
+
+validate_json_schema("/path/to/schema.json", data_path="/path/to/some/json", msg="Some helpful message should this validation fail")
+
+# indent should pretty print the json contents of the error to make it
+# more easily parsed by humans
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary, indent=2)
+"""
+
+
+def validate_json_schema(
+    schema_path: str,
+    data_dict_or_path: Union[Dict, str],
+    msg: str = "",
+    indent: int = 2,
+) -> Optional[ValidationError]:
+    """
+    Validate metadata.json files against schema provided.
+
+    `schema_path`: file path of schema to validate against
+    `data_dict_or_path`: file path or dictionary of data to be validated
+    `msg`: optional string to provide additional information about validation
+    `indent`: optional integer to be used when indenting json output
+    """
+    # Load schema as dict
+    parsed_schema_path = urlparse(schema_path)
+    if parsed_schema_path.scheme == "http":
+        # TODO Load schema from URL
+        raise NotImplementedError("Validation from remote schema not yet supported")
+    else:
+        with open(Path(schema_path), "r") as f:
```

Comment:
turn it into a path then check the path exists first. Obvious well explained message if it does not.